### ISSUE TITLE
Docker via WSL2 doesn't do IPv4 loopback?

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonKUBE#968]
@johncburns1: I'd like to do a neonLIBRARY release of v2.2.0 so I can start a new release where the libraries target Netstandard2.1 (this is a good time for doing this).  The subsequent release will be v3.0.0 since this is a major breaking change.

I rebuilt and published all of the Docker images as part of the release process and am seeing these errors:
```
  Message: 
    System.AggregateException : One or more errors occurred. (Cannot connect to Cadence cluster.)
    ---- Neon.Cadence.ConnectException : Cannot connect to Cadence cluster.
    -------- Neon.Cadence.CadenceCustomException : Messages: Could not establish a connection with the cadence server.
  Stack Trace: 
    Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
    Task`1.GetResultCore(Boolean waitCompletionNotification)
    Task`1.get_Result()
    CadenceFixture.StartAsComposed(CadenceSettings settings, String image, String name, String[] env, String defaultDomain, LogLevel logLevel, Boolean reconnect, Boolean keepRunning, String hostInterface, Boolean noClient, Boolean noReset, ContainerLimits limits) line 311
    <>c__DisplayClass8_0.<Start>b__0() line 185
    TestFixture.Start(Action action) line 234
    CadenceFixture.Start(CadenceSettings settings, String image, String name, String[] env, String defaultDomain, LogLevel logLevel, Boolean reconnect, Boolean keepRunning, String hostInterface, Boolean noClient, Boolean noReset, ContainerLimits limits) line 169
    Test_EndToEnd.ctor(CadenceFixture fixture) line 84
    ----- Inner Stack Trace -----
    CadenceClient.ConnectAsync(CadenceSettings settings) line 973
    ----- Inner Stack Trace -----
    ProxyReply.ThrowOnError() line 90
    CadenceClient.ConnectAsync(CadenceSettings settings) line 968
```
Is it possible that you did a special manual build of the **cadence-dev** image and perhaps the normal image build script wasn't updated?

**Comment by jefflill** [comment]
Oh sorry, that could be it.  I’ll check when I get back home.  I thought I
had modified the cadence and temporal test fixtures to check for the other
in and stop it first.  Maybe that’s not working (or my memory is failing).

—Jeff

**Comment by johncburns1** [comment]
Yeah just let me know if it is still not working.  All tests are passing on my machine on the `release-neonLIBRARY-2.2.0` branch, but maybe we are not completely synced up.

**Comment by jefflill** [comment]
This appears to be Docker running on WSL2.  I just upgraded last week and this is probably the first time I tried running unit tests since then.  I suspect that the loopback interfaces are different for native Windows vs. WSL2.

**Comment by jefflill** [comment]
Looks like Docker is binding to the IPv6 loopback address, not the IPv4 127.0.0.1 address:

https://github.com/microsoft/WSL/issues/4983

**Comment by jefflill** [comment]
It looks like the solution may be to **COMPLETELY** uninstall Docker on both the Windows and WSL side, being sure to remove any remnants, and then reinstall Docker:

https://github.com/microsoft/WSL/issues/4321#issuecomment-573351391

**Comment by jefflill** [comment]
We addressed this by using localhost instead of 127.0.0.1.  This works because DNS lookup of localhost returns the IPv6 loop back address which works for WSL2.
